server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java

```diff
@@ -515,13 +518,15 @@ protected FileCompactorRunnable createCompactionJob(final TExternalCompactionJob
 
     return new FileCompactorRunnable() {
 
-      private AtomicReference<FileCompactor> compactor = new AtomicReference<>();
+      private final AtomicReference<FileCompactor> compactor = new AtomicReference<>();
+      private long startTimeNanos = -1;
```

Comment:
I think multiple threads are reading/writing this.  Should probably make it volatile or use AtomicLong